### PR TITLE
feat(app): Add NATS RPC Example

### PR DESCRIPTION
**Pull request by AlexCuse**
Adds example of a custom synchronous server using NATS request/reply
semantics through the demo.nats.io broker.

Signed-off-by: Alex Ullrich <alex.ullrich@gmail.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)

### CODE REVIEW

File: application-services/custom/custom-trigger-nats-rpc/main.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	// turn off secure mode for examples. Not recommended for production
+	_ = os.Setenv("EDGEX_SECURITY_SECRET_STORE", "false")
+
+	// First thing to do is to create an instance of the EdgeX SDK Service, which also runs the bootstrap initialization.
+	service, ok := pkg.NewAppServiceWithTargetType(serviceKey, &[]byte{})
+	if !ok {
+		os.Exit(-1)
+	}
+
+	service.RegisterCustomTriggerFactory("custom-stdin", func(config interfaces.TriggerConfig) (interfaces.Trigger, error) {
```
Comment (lenny-intel) [inline application-services/custom/custom-trigger-nats-rpc/main.go:73]:
```suggestion
	service.RegisterCustomTriggerFactory("custom-rpc", func(config interfaces.TriggerConfig) (interfaces.Trigger, error) {
```

File: application-services/custom/custom-trigger-nats-rpc/res/configuration.toml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  Port = 8200
+  Path = "app-custom-trigger/"
+  TokenFile = "/tmp/edgex/secrets/app-custom-trigger/secrets-token.json"
+  RootCaCertPath = ""
+  ServerName = ""
+  [SecretStore.Authentication]
+    AuthType = "X-Vault-Token"
+
+[Trigger]
+Type="custom-stdin"
```
Comment (lenny-intel) [inline application-services/custom/custom-trigger-nats-rpc/res/configuration.toml:38]:
```suggestion
Type="custom-rpc"
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Comment by AlexCuse** [comment]
Good catch @lenny-intel - could almost correct it through github now but it didn't put DCO on the generated commit :)

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**